.circleci/config.yml

```diff
@@ -0,0 +1,135 @@
+version: 2.1
+
+defaults: &defaults
+  environment:
+    OPAMVERBOSE: 1
+    OPAMWITHTEST: true
+    OPAMYES: true
+    TERM: xterm
+
+commands:
+  startup:
+    steps:
+    - checkout
+    - run:
+        name: Pull submodules
+        command: git submodule update --init
+    - run:
+        name: Configure environment
+        command: echo . ~/.profile >> $BASH_ENV
+    - run:
+        name: Install dependencies
+        command: |
+          opam repo -a add coq-extra-dev https://coq.inria.fr/opam/extra-dev
+          opam update
+          opam pin add --kind=path . -n  # Make opam aware of our dependency constraints
+          opam install --deps-only .
+    - run:
+        name: List installed packages
+        command: opam list
+  build:
+    steps:
+    - run:
+        name: Build and test locally
+        command: |
+          make
+          make test
+    - run:
+        name: Build, test, and install package
+        command: opam install .
+    - run:
+        name: Uninstall package
+        command: opam uninstall .
+    - run:
+        name: Cleanup local directory
+        command: |
+          make clean
+          ls -AGR
+  deploy:
+    steps:
+    - run:
+        name: Build Documentation
+        command: make html
+    - run:
+        name: Configure Git
+        command: |
+          git config --global core.autocrlf input
+          git config --global user.name  "Yishuai Li"
+          git config --global user.email "yishuai@cis.upenn.edu"
```

Comment:
Hmm I'm signing all automatic commits to `gh-pages` branch.

Reply:
Ah, not a problem yet as you didn't `deploy` in workflow.

Reply:
:+1: I might still want to remove it to avoid accidents...